sysutils/monit/src/opnsense/service/templates/OPNsense/Monit/monitrc

```diff
@@ -2,20 +2,60 @@
 
 {% from 'OPNsense/Macros/interface.macro' import physical_interface %}
 {% if helpers.exists('OPNsense.monit.general') %}
+{%  if helpers.exists('OPNsense.monit.general.httpdEnabled') and OPNsense.monit.general.httpdEnabled|default('0') == '1' %}
+{%  set httpdCredentials = OPNsense.monit.general.httpdUsername ~ ':"' ~ OPNsense.monit.general.httpdPassword ~ '"' %}
+{%    set httpdport = "port " ~ OPNsense.monit.general.httpdPort %}
+set httpd unixsocket /var/run/monit.sock {{ httpdport }}
+    allow localhost
+    allow {{  httpdCredentials }}
+{%    if helpers.exists('OPNsense.monit.general.httpdAllow') %}
+{%      for allow in OPNsense.monit.general.httpdAllow.split(",") %}
+{%        set cred = allow.split(":") %}
+{%        if cred[1] %}
+{%          set pass = cred[1].split() %}
+{%          set allow = cred[0] ~ ':"' ~ pass[0] ~ '" ' ~ pass[1] %}
+{%        endif %}
+    allow {{ allow }}
+{%      endfor %}
+{%    endif %}
+{%    if helpers.exists('OPNsense.monit.general.mmonitUrl') and OPNsense.monit.general.mmonitUrl|default('') != '' %}
+{%      set mregister = 'register without credentials' if OPNsense.monit.general.mmonitRegisterCredentials|default('1') == '0' %}
+
+set mmonit {{ OPNsense.monit.general.mmonitUrl }} timeout {{ OPNsense.monit.general.mmonitTimeout }} seconds {{ mregister }}
+{%    endif %}
+{%  else %}
 set httpd unixsocket /var/run/monit.sock
-     allow localhost
+    allow localhost
+{%  endif %}
 
 set daemon {{ OPNsense.monit.general.interval }} with start delay {{ OPNsense.monit.general.startdelay }}
 
+{%  if helpers.exists('OPNsense.monit.general.logfile') %}
```

Comment:
not entirely sure, but it is probably safer to additionally check if the value is not empty in case the user (and the model) saves it that way after an edit